### ISSUE TITLE
HaltonFrame

### ISSUE THREAD

**Issue by paul-vdb** [javalake2023/spbal#3]
Default is N=1 for sample size. However, there are catches for is.null(N) for choosing the whole frame vs not.

```
  if(base::is.null(N)){
    wantHaltonGrid <- TRUE
    wantHaltonFrame <- FALSE
    if(verbose){
      base::message("spbal(HaltonFrame) Request for a Halton Grid.")
    }
  }
```

Behaviour is strange that N= 1 returns all clipped points from the grid, but N=2 returns 2. Giving N = NULL breaks the code.
![image](https://github.com/javalake2023/spbal/assets/13092582/557d5987-60b1-4e41-9ee3-c98368e130d1)

I think default of NULL and return everything is fine, but default of 1 and return everything doesn't make sense. If someone wants 1 point, they should get one point.

**Comment by paul-vdb** [comment]
Why is this being called?
`pts.shp <- rotate.scale.coords(coords = pts, bb = bb.new)` in `getHaltonFrame`. We aren't rotating and haven't done anything in that space yet, so it seems unnecessary. I remember in a meeting we talked about how it made more sense to just rotate the shape before calling the function and then rotating the samples back onto the shape after generating it.

**Comment by paul-vdb** [comment]
If the user asks for more points in the Halton Frame than exist within the polygon, it appears as if the grid is automatically intensified. I assume we would prefer to tell the user to choose a different grid and stop at that point.

**Comment by paul-vdb** [comment]
If the user asks for 25 points on a dense Halton Frame, we clip the entire frame before we subset for those 25 points. This seems like a lot of computation when not required. Couldn't we just iterate until we have found the desired number of points?

I'm confused why in `getHaltonPointsFromExpandableGrid` it has to ensure that the first point falls into the study region `stratify_found_first == FALSE`. Shouldn't this be enforced when we set the seed like in BAS? Once we have a seed, then we should be able to just call the Halton indices directly. The user should then be able to use that same seed to call the function again to get the same result, but faster the second time given the first frame was in the study region was already enforced.
